### PR TITLE
Add control nodes and compute nodes specific alarms

### PR DESCRIPTION
**Pull request by elemoine**
This commits overrides CPU triggers defined in the Linux formula. This is to use specific rules on OpenStack control and compute nodes.

### CODE REVIEW

**Comment by elemoine** [comment]
> to be able to have aggregation per node "role"

Good call. I'll see what I can do.

**Comment by elemoine** [comment]
@SwannCroiset I created an alarm cluster for "control" nodes and another one for "compute" nodes. Note that this requires overriding all the system alarms defined in the Linux formula, to be able to tag the alarm metrics with the `node_role`.

File: classes/cluster/mk20_stacklight_basic/openstack/compute.yml [hunk 1]
```diff
@@ -28,6 +28,102 @@ parameters:
...
+          - linux_system_network_warning_dropped_tx
+          dimension:
+            node_role: compute
+        linux_system_hdd_errors:
+          alerting: enabled_with_notification
+          triggers:
+          - linux_system_hdd_errors_critical
+          dimension:
+            node_role: compute
+    aggregator:
```
Comment (SwannCroiset) [inline classes/cluster/mk20_stacklight_basic/openstack/compute.yml]: I'm wondering how we are going to aggregate this GSE node with the top-clusters aggregation

File: classes/cluster/mk20_stacklight_basic/openstack/compute.yml [hunk 2]
```diff
@@ -28,6 +28,102 @@ parameters:
...
+      alarm_cluster:
+        compute_nodes:
+          policy: majority_of_members
```
Comment (SwannCroiset) [inline classes/cluster/mk20_stacklight_basic/openstack/compute.yml]: you should explicitly set:
`alerting: enabled_with_notification`

File: classes/cluster/mk20_stacklight_basic/openstack/control.yml [hunk 3]
```diff
@@ -34,6 +34,114 @@ parameters:
...
+        linux_system_hdd_errors:
+          alerting: enabled_with_notification
+          triggers:
+          - linux_system_hdd_errors_critical
+          dimension:
+            node_role: control
+    aggregator:
+      alarm_cluster:
+        control_nodes:
+          policy: majority_of_members
```
Comment (SwannCroiset) [inline classes/cluster/mk20_stacklight_basic/openstack/control.yml]: you should explicitly set:
`alerting: enabled_with_notification`

File: classes/cluster/mk20_stacklight_basic/openstack/compute.yml (hunk 1 above)
Comment (elemoine) [inline classes/cluster/mk20_stacklight_basic/openstack/compute.yml]: You would need to create your own "top top level cluster" for that. I don't know what else to say…

File: classes/cluster/mk20_stacklight_basic/openstack/compute.yml (hunk 2 above)
Comment (elemoine) [inline classes/cluster/mk20_stacklight_basic/openstack/compute.yml]: Done.

File: classes/cluster/mk20_stacklight_basic/openstack/control.yml (hunk 3 above)
Comment (elemoine) [inline classes/cluster/mk20_stacklight_basic/openstack/control.yml]: Done.

**Comment by elemoine** [comment]
I added [WIP] to the PR title because this requires https://github.com/Mirantis/mk-lab-salt-model/pull/99, which should be merged first.

**Comment by cznewt** [comment]
can this be defined on system level and referenced from cluster? system.heka.metric_collector.alarm.openstack_compute

**Comment by elemoine** [comment]
> can this be defined on system level and referenced from cluster? system.heka.metric_collector.alarm.openstack_compute

Yep, possibly. I'll look into it.

**Comment by elemoine** [comment]
> can this be defined on system level and referenced from cluster? system.heka.metric_collector.alarm.openstack_compute

Done. Good suggestion.

**Pull request merged**